File: ios/Plugin/CapacitorWifiConnectPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(CapacitorWifiConnectPlugin, "CapacitorWifiConnect",
          CAP_PLUGIN_METHOD(checkPermission, CAPPluginReturnPromise);
          CAP_PLUGIN_METHOD(requestPermission, CAPPluginReturnPromise);
          CAP_PLUGIN_METHOD(disconnect, CAPPluginReturnPromise);
          CAP_PLUGIN_METHOD(getAppSSID, CAPPluginReturnPromise);
          CAP_PLUGIN_METHOD(getDeviceSSID, CAPPluginReturnPromise);
          CAP_PLUGIN_METHOD(connect, CAPPluginReturnPromise);
          CAP_PLUGIN_METHOD(prefixConnect, CAPPluginReturnPromise);
          CAP_PLUGIN_METHOD(secureConnect, CAPPluginReturnPromise);
          CAP_PLUGIN_METHOD(securePrefixConnect, CAPPluginReturnPromise);
)

File: ios/Plugin/CapacitorWifiConnectPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(CapacitorWifiConnectPlugin)
public class CapacitorWifiConnectPlugin: CAPPlugin {
    private let implementation = CapacitorWifiConnect()

    @objc func checkPermission(_ call: CAPPluginCall) {
        implementation.checkPermission(resolve: call.resolve);
    }
    
    @objc func requestPermission(_ call: CAPPluginCall) {
        implementation.requestPermission(resolve: call.resolve, reject: call.reject);
    }
    
    @objc func disconnect(_ call: CAPPluginCall) {
        implementation.disconnect(resolve: call.resolve, reject: call.reject);
    }
    
    @objc func getAppSSID(_ call: CAPPluginCall) {
        implementation.getAppSSID(resolve: call.resolve, reject: call.reject);
    }
    
    @available(iOS 14.0, *)
    @objc func getDeviceSSID(_ call: CAPPluginCall) {
        implementation.getDeviceSSID(resolve: call.resolve, reject: call.reject);
    }
    
    @objc func connect(_ call: CAPPluginCall) {
        let ssid = call.getString("ssid");
        let saveNetwork = call.getBool("saveNetwork") ?? false;
        if(ssid != nil) {
            implementation.connect(ssid: ssid!, saveNetwork: saveNetwork, resolve: call.resolve, reject: call.reject);
        } else {
            call.reject("SSID is mandatory")
        }
    }
    
    @objc func prefixConnect(_ call: CAPPluginCall) {
        let ssid = call.getString("ssid");
        let saveNetwork =  call.getBool("saveNetwork") ?? false;
        if(ssid != nil) {
      
            implementation.prefixConnect(ssid: ssid!, saveNetwork: saveNetwork, resolve: call.resolve, reject: call.reject);
        } else {
            call.reject("SSID is mandatory")
        }
    }
    
    @objc func secureConnect(_ call: CAPPluginCall) {
        let ssid = call.getString("ssid");
        let password = call.getString("password");
        let saveNetwork = call.getBool("saveNetwork") ?? false;
        let isWep = call.getBool("isWep") ?? false;
        if(ssid != nil && password != nil) {
            implementation.secureConnect(ssid: ssid!, password: password!, saveNetwork: saveNetwork, isWep: isWep, resolve: call.resolve, reject: call.reject);
        } else {
            call.reject("SSID and password are mandatory")
        }
    }
    
    @objc func securePrefixConnect(_ call: CAPPluginCall) {
        let ssid = call.getString("ssid");
        let password = call.getString("password");
        let saveNetwork = call.getBool("saveNetwork") ?? false;
        let isWep = call.getBool("isWep") ?? false;
        if(ssid != nil && password != nil) {
            
            implementation.securePrefixConnect(ssid:ssid!, password:password!, saveNetwork:saveNetwork, isWep: isWep, resolve: call.resolve, reject: call.reject);
        } else {
            call.reject("SSID and password are mandatory")
        }
    }
}
